### ISSUE TITLE
64-bit Linux Ubuntu 17.04 rclone v1.39-175-g25bbc5d2β segfault

### ISSUE THREAD

**Issue by jikamens** [ncw/rclone#2132]
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x9f1017]

goroutine 1503 [running]:
github.com/ncw/rclone/backend/local.(*localOpenFile).Read(0xc4207a8030, 0xc4209a802d, 0xffd3, 0xffe3, 0x2d, 0x0, 0x0)
	/home/travis/gopath/src/github.com/ncw/rclone/backend/local/local.go:661 +0x47
io.ReadAtLeast(0x7f76684d1ad8, 0xc4207a8030, 0xc4209a8000, 0x10000, 0x10010, 0x10000, 0xc42024e200, 0xc42024e220, 0xc42091d7e8)
	/home/travis/.gimme/versions/go1.10.linux.amd64/src/io/io.go:309 +0x86
io.ReadFull(0x7f76684d1ad8, 0xc4207a8030, 0xc4209a8000, 0x10000, 0x10010, 0x73183d, 0x158ea90, 0xc42024e200)
	/home/travis/.gimme/versions/go1.10.linux.amd64/src/io/io.go:327 +0x58
github.com/ncw/rclone/backend/crypt.(*encrypter).Read(0xc4205281b0, 0xc4208a6240, 0x65, 0xdc0, 0x0, 0x0, 0x0)
	/home/travis/gopath/src/github.com/ncw/rclone/backend/crypt/cipher.go:666 +0xf0
github.com/ncw/rclone/fs/accounting.(*Account).read(0xc420140780, 0x1019080, 0xc4205281b0, 0xc4208a6240, 0x65, 0xdc0, 0x20, 0x0, 0x0)
	/home/travis/gopath/src/github.com/ncw/rclone/fs/accounting/accounting.go:141 +0x107
github.com/ncw/rclone/fs/accounting.(*accountStream).Read(0xc4203540e0, 0xc4208a6240, 0x65, 0xdc0, 0x20, 0x0, 0x0)
	/home/travis/gopath/src/github.com/ncw/rclone/fs/accounting/accounting.go:317 +0x5e
io.(*teeReader).Read(0xc4203b7da0, 0xc4208a6240, 0x65, 0xdc0, 0x20, 0x0, 0x0)
	/home/travis/.gimme/versions/go1.10.linux.amd64/src/io/io.go:533 +0x55
github.com/ncw/rclone/backend/b2.(*hashAppendingReader).Read(0xc420c8e4c0, 0xc4208a6240, 0x65, 0xdc0, 0x20, 0x0, 0x0)
	/home/travis/gopath/src/github.com/ncw/rclone/backend/b2/upload.go:36 +0x7c
net/http.transferBodyReader.Read(0xc4203ac3c0, 0xc4208a6240, 0x65, 0xdc0, 0x20, 0x0, 0x0)
	/home/travis/.gimme/versions/go1.10.linux.amd64/src/net/http/transfer.go:60 +0x56
io.(*LimitedReader).Read(0xc420384bc0, 0xc4208a6240, 0xdc0, 0xdc0, 0x20, 0x0, 0x0)
	/home/travis/.gimme/versions/go1.10.linux.amd64/src/io/io.go:446 +0x63
bufio.(*Writer).ReadFrom(0xc42097e0c0, 0x1019be0, 0xc420384bc0, 0x7f76684410a8, 0xc42097e0c0, 0x1)
	/home/travis/.gimme/versions/go1.10.linux.amd64/src/bufio/bufio.go:703 +0xcd
io.copyBuffer(0x1018c80, 0xc42097e0c0, 0x1019be0, 0xc420384bc0, 0x0, 0x0, 0x0, 0xe2f7c0, 0xc4202c6101, 0xc420384bc0)
	/home/travis/.gimme/versions/go1.10.linux.amd64/src/io/io.go:386 +0x31a
io.Copy(0x1018c80, 0xc42097e0c0, 0x1019be0, 0xc420384bc0, 0xc4208a621e, 0xf2a248, 0x2)
	/home/travis/.gimme/versions/go1.10.linux.amd64/src/io/io.go:362 +0x5a
net/http.(*transferWriter).WriteBody(0xc4203ac3c0, 0x1018c80, 0xc42097e0c0, 0x2, 0x2)
	/home/travis/.gimme/versions/go1.10.linux.amd64/src/net/http/transfer.go:337 +0x5fa
net/http.(*Request).write(0xc420750000, 0x1018c80, 0xc42097e0c0, 0x0, 0xc42051c780, 0x0, 0x0, 0x0)
	/home/travis/.gimme/versions/go1.10.linux.amd64/src/net/http/request.go:622 +0x6b2
net/http.(*persistConn).writeLoop(0xc4201b3200)
	/home/travis/.gimme/versions/go1.10.linux.amd64/src/net/http/transport.go:1825 +0x1ea
created by net/http.(*Transport).dialConn
	/home/travis/.gimme/versions/go1.10.linux.amd64/src/net/http/transport.go:1238 +0x97f
```

**Comment by jikamens** [comment]
FYI, I (so far) can't reproduce this with the current beta (v1.39-232-g4e90ad04β), so this may be moot. But I thought I should report it anyway just in case.

**Comment by ncw** [comment]
This was fixed in 7d4da1c66ad434d85452b311f6acd36ca8727037 so you shouldn't have any problems with the latest beta.

Thanks for reporting though - it might have been a new issue!

**Comment by ncw** [comment]
I'll close this now.
